drivers/i2c/CMakeLists.txt

```diff
@@ -6,107 +6,99 @@ zephyr_library()
 
 zephyr_library_sources(i2c_common.c)
 
+zephyr_library_sources_ifdef(CONFIG_GPIO_I2C_SWITCH	gpio_i2c_switch.c)
```

Comment:
Please consider to add.
```
# zephyr-keep-sorted-start
```

Such like https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/gpio/Kconfig#L97